hyrisecockpit/api/app/database/service.py

```diff
@@ -62,22 +63,29 @@ def deregister_database(cls, interface: DatabaseInterface) -> int:
         return response["header"]["status"]
 
     @classmethod
-    def get_available_benchmark_tables(cls) -> AvailableBenchmarkTables:
-        """Return all available benchmarks."""
-        return AvailableBenchmarkTables(
-            folder_names=["tpch_0.1", "tpch_1", "tpcds_1", "job"]
-        )
+    def get_available_workload_tables(cls) -> AvailableWorkloadTables:
+        """Return all available workloads."""
+        drivers = Connector.get_workload_drivers()
+        workload_tables = []
+        for workload_type, driver in drivers.items():
+            for scale_factor in driver.scale_factors:
+                workload_tables.append(
+                    WorkloadTables(
+                        workload_type=workload_type, scale_factor=scale_factor
+                    )
+                )
+        return AvailableWorkloadTables(workload_tables=workload_tables)
```

Comment:
See [PEP 202](https://www.python.org/dev/peps/pep-0202/):
> List comprehensions provide a more concise way to create lists in situations where [...] nested loops would currently be used.

```suggestion
        return AvailableWorkloadTables(
            workload_tables=[
                WorkloadTables(workload_type=workload_type, scale_factor=scale_factor)
                for workload_type, driver in drivers.items()
                for scale_factor in driver.scale_factors
            ]
        )
```